### PR TITLE
feat: CP support Typography className and style

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | feat: CP support Typography className and style |
| 🇨🇳 Chinese | CP 支持 Typography 的 className 和 style 属性 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 3a78d94</samp>

This pull request adds a new `typography` prop to the `ConfigProvider` component that allows users to customize the global typography configuration for the ant-design components. It also updates the `Typography` component to respect the ConfigProvider settings and the corresponding test and documentation files.

### <samp>🤖 Generated by Copilot at 3a78d94</samp>

*  Add a new typography prop to the ConfigProvider component and the ConfigContext to allow customizing the className and style of the Typography component ([link](https://github.com/ant-design/ant-design/pull/43070/files?diff=unified&w=0#diff-f7a3a5082a18fdd0627b75fb4f13fa559a2e5a5a781e381b8ba8c98b13318ccaL90-R99), [link](https://github.com/ant-design/ant-design/pull/43070/files?diff=unified&w=0#diff-0a9895a32672ec55a084aa7165fc8666008942903b8e51e30b662e9e1ecffb56R139-R142), [link](https://github.com/ant-design/ant-design/pull/43070/files?diff=unified&w=0#diff-0a9895a32672ec55a084aa7165fc8666008942903b8e51e30b662e9e1ecffb56R230), [link](https://github.com/ant-design/ant-design/pull/43070/files?diff=unified&w=0#diff-0a9895a32672ec55a084aa7165fc8666008942903b8e51e30b662e9e1ecffb56R282))
*  Modify the Typography component to receive the typography prop from the ConfigContext and apply it to its own props ([link](https://github.com/ant-design/ant-design/pull/43070/files?diff=unified&w=0#diff-b876734b2f41ebc0a47dca5b8a06d9cfd09a0489c33414ab138415a648854fe6L4-R6), [link](https://github.com/ant-design/ant-design/pull/43070/files?diff=unified&w=0#diff-b876734b2f41ebc0a47dca5b8a06d9cfd09a0489c33414ab138415a648854fe6L32-R86))
*  Import the Typography component and add a test case to check if it can receive the typography prop from the ConfigProvider component ([link](https://github.com/ant-design/ant-design/pull/43070/files?diff=unified&w=0#diff-0240afe2faa6807b920ad4c86a45d6724eb47ed4cde840f097b69fdb4d28cdb5R12), [link](https://github.com/ant-design/ant-design/pull/43070/files?diff=unified&w=0#diff-0240afe2faa6807b920ad4c86a45d6724eb47ed4cde840f097b69fdb4d28cdb5R202-R213))
*  Add the typography entry to the API documentation of the config-provider component in both English and Chinese ([link](https://github.com/ant-design/ant-design/pull/43070/files?diff=unified&w=0#diff-975af6d8b4e359c4e88a7586cf1c91b35989cdadc9f2dc9ca16a376b5f5b0563R73), [link](https://github.com/ant-design/ant-design/pull/43070/files?diff=unified&w=0#diff-b6a7375df96aaccf21f7452705af80f8cd8c7323f0fc2b9dbdc71500b8478775R74))
*  Improve the code readability of the getPrefixCls function in the ConfigProvider component by adding curly braces and a new line to the if statement ([link](https://github.com/ant-design/ant-design/pull/43070/files?diff=unified&w=0#diff-0a9895a32672ec55a084aa7165fc8666008942903b8e51e30b662e9e1ecffb56L242-R249))

### CODE REVIEW

File: components/typography/Typography.tsx [hunk 1]
```diff
@@ -29,53 +29,60 @@ interface InternalTypographyProps<C extends keyof JSX.IntrinsicElements>
...
-    const componentClassName = classNames(
-      prefixCls,
-      {
-        [`${prefixCls}-rtl`]: direction === 'rtl',
-      },
-      className,
-      rootClassName,
-      hashId,
-    );
+  const mergeStyle = React.useMemo<React.CSSProperties>(
```
Comment (zombieJ) [inline components/typography/Typography.tsx:74]: 这个 `useMemo` 没有必要，性能可能反而会有影响。

File: components/config-provider/index.en-US.md [hunk 2]
```diff
@@ -70,6 +70,7 @@ Some components use dynamic style to support wave effect. You can config `csp` p
 | space | Set Space common props, ref [Space](/components/space) | { size: `small` \| `middle` \| `large` \| `number`, className?: string, style?: React.CSSProperties, classNames?: { item: string }, styles?: { item: React.CSSProperties } } | - | 5.6.0 |
 | theme | Set theme, ref [Customize Theme](/docs/react/customize-theme) | - | - | 5.0.0 |
 | virtual | Disable virtual scroll when set to `false` | boolean | - | 4.3.0 |
+| typography | Set Typography common props | { className?: string, style?: React.CSSProperties } | - | 5.7.0 |
```
Comment (MadCcc) [inline components/config-provider/index.en-US.md:73]: 按字母顺序排一下哈

File: components/typography/Typography.tsx (hunk 1 above)
Comment (li-jia-nan) [reply components/typography/Typography.tsx]: done~

**Comment by MadCcc** [comment]
https://github.com/ant-design/ant-design/pull/43042/files#diff-b6a7375df96aaccf21f7452705af80f8cd8c7323f0fc2b9dbdc71500b8478775
参照一下最新代码，新增了统一的测试文件和 API 表格，按照这个改一下？

File: components/typography/Typography.tsx [hunk 3]
```diff
@@ -29,53 +29,57 @@ interface InternalTypographyProps<C extends keyof JSX.IntrinsicElements>
...
-    const componentClassName = classNames(
-      prefixCls,
-      {
-        [`${prefixCls}-rtl`]: direction === 'rtl',
-      },
-      className,
-      rootClassName,
-      hashId,
-    );
+  const mergeStyle: React.CSSProperties = { ...typography?.style, ...style };
```
Comment (MadCcc) [inline components/typography/Typography.tsx:74]:
```suggestion
  const mergedStyle: React.CSSProperties = { ...typography?.style, ...style };
```
捉个小虫

**Comment by li-jia-nan** [comment]
> https://github.com/ant-design/ant-design/pull/43042/files#diff-b6a7375df96aaccf21f7452705af80f8cd8c7323f0fc2b9dbdc71500b8478775 参照一下最新代码，新增了统一的测试文件和 API 表格，按照这个改一下？

done~

File: components/config-provider/index.zh-CN.md [hunk 4]
```diff
@@ -108,6 +108,7 @@ const {
 | form | 设置 Form 组件的通用属性 | { validateMessages?: [ValidateMessages](/components/form-cn#validatemessages), requiredMark?: boolean \| `optional`, colon?: boolean, scrollToFirstError?: boolean \| [Options](https://github.com/stipsan/scroll-into-view-if-needed/tree/ece40bd9143f48caf4b99503425ecb16b0ad8249#options)} | - | requiredMark: 4.8.0; colon: 4.18.0; scrollToFirstError: 5.2.0 |
 | input | 设置 Input 组件的通用属性 | { className?: string, style?: React.CSSProperties } | - | 5.7.0 |
 | select | 设置 Select 组件的通用属性 | { showSearch?: boolean } | - |  |
+| typography | 设置 Typography 组件的通用属性 | { className?: string, style?: React.CSSProperties } | - | 5.7.0 |
```
Comment (MadCcc) [inline components/config-provider/index.zh-CN.md:111]: 写在下一行哈

Comment (li-jia-nan) [reply components/config-provider/index.zh-CN.md]: done~

**Review by MadCcc — APPROVED** [review]

**Pull request merged**